src/branding/TerraHexagonsAnimation.ts

```diff
@@ -91,3 +90,31 @@ export const TerraHexagonsAnimation = (props: TerraHexagonsAnimationProps): Reac
     ]
   );
 };
+
+export interface TerraHexagonsAnimationOverlayProps extends TerraHexagonsAnimationProps {
+  message: string;
+}
+
+export const TerraHexagonsAnimationOverlay = (props: TerraHexagonsAnimationOverlayProps): ReactNode => {
```

Comment:
I think you should make a separate file for this component (TSX) and name it something that makes it clear how it is intended to be used. For instance, `TerraHexagonsLengthyOperationOverlay` or simply `LengthyOperationOverlay`… after all, the fact that it has TerraHexagons in it currently is an implementation detail that could change.

Reply:
I also suggest putting it in `src\components`.

Reply:
Do you have an opinion about where the tsx should go? Maybe `components/src/spinners`?

Reply:
Oh, but then it can't use `TerraHexagonsAnimation`

Reply:
I wouldn't suggest putting it in the shared components package because the current implementation is very Terra-specific. So just `src/components`.

Reply:
Since its colors directly rely on the colors in `TerraHexagonsAnimation` I think it's best to keep it in the same directory. But I could move it if you have a strong opinion; either way would work.